File: app/src/main/java/extensions/java/util/ArrayList/ArrayListExtensions.java
package extensions.java.util.ArrayList;

import java.util.ArrayList;

import manifold.ext.rt.api.Extension;
import manifold.ext.rt.api.This;

@Extension
public class ArrayListExtensions {
    public static ArrayList<String> convertEmptyDataToDefault(@This ArrayList<String> thiz) {
        for (int i = 0; i < thiz.size(); i++) {
            if (thiz[i].equals("")) {
                thiz[i] = "0,0,0";
            }
        }
        return thiz;
    }

    public static ArrayList<String> equalizeSensorList(@This ArrayList<String> thiz, int maxLength, String lineToAdd) {
        int i = thiz.size();
        while (i < maxLength) {
            thiz.add(lineToAdd);
            i++;
        }
        return thiz;
    }
}

File: app/src/main/java/com/example/gyrocollector/helpers/ArrayListExtensions.kt
package com.example.gyrocollector.helpers

fun ArrayList<String>.convertEmptyDataToDefault(): ArrayList<String> {
    for (i in this.indices) {
        if (this[i] == "") {
            this[i] = "0,0,0"
        }
    }
    return this
}

fun ArrayList<String>.equalizeSensorList(
    maxLength: Int,
    lineToAdd: String
): ArrayList<String> {
    var i = this.size
    while (i < maxLength) {
        this.add(lineToAdd)
        i++
    }
    return this
}